lib/table.rb

```diff
@@ -0,0 +1,23 @@
+require_relative 'row'
+
+class Table
+  def initialize(node)
+    @table = node
+  end
+
+  def to_a
+    constituency = nil
+    constituency_id = nil
+    table.xpath('.//tr[td]').map do |tr|
+      tds = tr.xpath('./td')
+      next if tds.count == 1
+      constituency_id = tds.shift.text.strip if tds.count > 2
+      constituency = tds.shift.text.strip if tds.count > 2
```

Comment:
Rather than having all this logic in-place, which is likely to just grow more and more complex over time, I think this would be much simpler and much clearer with separate Table classes, each representing a different layout.